core/trino-main/src/main/java/io/trino/execution/executor/TaskExecutor.java

```diff
@@ -561,6 +561,15 @@ public void run()
                         }
                         splitFinished(split);
                     }
+                    finally {
+                        // Clear the interrupted flag on the current thread, driver cancellation may have triggered an interrupt
+                        if (Thread.interrupted()) {
```

Comment:
What is the motivation of this change?

Based on my understanding terminating a runner tread upon interruption is by design (see line 503, `while (!closed && !Thread.currentThread().isInterrupted()) {`)

Reply:
Terminating a runner isn't necessary if the `TaskExecutor` is not closed. Essentially there are two sources of interrupts that can occur:
1. `TaskExecutor#shutdown()` sets `closed = true`, calls `executor.shutdownNow()` which will interrupt all threads. If they're blocked in a call to `waitingSplits.take()`, they will wake up and throw `InterruptedException` and the runner will terminate (without adding a replacement for itself). This happens exactly once during the whole lifecycle of the JVM in production, but in tests obviously it can happen multiple times.
2. The active driver is closed while the `TaskRunner` is running it. The task runner thread is interrupted and may or may not throw an `InterruptedException` as a result since it's in the middle of a call to `split.process()` and we can't know for sure that an interruptable method will be called. Alternatively, some code inside of the driver interrupts the current thread for some other reason. Regardless of when/why that happens, the runner will terminate on the next iteration if we leave the flag set (because of `!Thread.currentThread().isInterrupted()`) but if `TaskExecutor#shutdown()` was _not_ called then there's no need to let that happen since it will just re-queue a new `TaskRunner` to replace itself via line 576 in the finally block.

So we can see that there are essentially two points to interrupting the thread: 1. stop all `TaskRunner`s because we're shutting down (rare) or 2. stop a single driver that happens to be running on a given `TaskRunner`. In the second case, pushing a new runner to replace the one that will terminate into the executor will create a new thread (because it's a cached threadpool) if an extra thread doesn't already exist there, which isn't necessary and does have a real runtime memory and performance cost.